File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    info = list(zip(deliveries,pickups))[::-1]
    
    tmp_d,tmp_p = 0,0
    for i in range(n):
        tmp_d += info[i][0]
        tmp_p += info[i][1]
        while tmp_d > 0 or tmp_p > 0:
            tmp_d -= cap
            tmp_p -= cap
            answer += (n-i) * 2
    
    return answer

File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.js
function solution(cap, n, deliveries, pickups) {
    let answer = 0;
    let r_deliveries = deliveries.reverse()
    let r_pickups = pickups.reverse()
    
    let tmp_d = 0
    let tmp_p = 0
    
    for(var i=0; i<n; i++){
        tmp_d += r_deliveries[i]
        tmp_p += r_pickups[i]
        while(tmp_d>0 || tmp_p >0){
            tmp_d -= cap
            tmp_p -= cap
            answer += (n-i) * 2
        }
    }
    return answer;
}